flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala

```diff
@@ -783,6 +783,10 @@ class TemporalTypesTest extends ExpressionTestBase {
     testSqlApi(timestampTz("2018-03-14 19:01:02.123"), "2018-03-14 19:01:02.123")
     testSqlApi(timestampTz("2018-03-14 19:00:00.010"), "2018-03-14 19:00:00.01")
 
+    testSqlApi(
+      timestampTz("2018-03-14 19:00:00.010") + " > " + timestampTz("2018-03-14 19:00:00.012"),
```

Comment:
It seems that this test doesn't reproduce the bug. Could use replace one of the constants to a field reference?